### PR TITLE
Serve docs from the server

### PR DESCRIPTION
**Pull request by svix-dylan**
Includes our OpenAPI documentation as static files, and serves them
from the Rust server.

### CODE REVIEW

**Comment by svix-dylan** [comment]
Open to suggestions on where these files and this module should live. I thought about putting them in the `/endpoints/v1` folder, but since they're available at `/docs` I don't feel great putting them in a directory that represents a nested path.

File: server/svix-server/src/lib.rs [hunk 1]
```diff
@@ -135,3 +136,27 @@ pub async fn run(cfg: Configuration, listener: Option<TcpListener>) {
...
+    use axum::{
+        response::{Html, IntoResponse},
+        routing::get,
+        Json, Router,
+    };
+
+    pub fn router() -> Router {
+        Router::new()
+            .route("/docs", get(get_docs))
+            .route("/openapi.json", get(get_openapi_json))
```
Comment (tasn) [inline server/svix-server/src/lib.rs:150]:
```suggestion
            .route("/api/v1/openapi.json", get(get_openapi_json))
```
Just to be compatible with what we have in prod.

File: server/svix-server/src/lib.rs [hunk 2]
```diff
@@ -135,3 +136,27 @@ pub async fn run(cfg: Configuration, listener: Option<TcpListener>) {
...
+    }
+
+    async fn get_docs() -> Html<&'static str> {
+        Html(include_str!("static/docs.html"))
+    }
+
+    async fn get_openapi_json() -> impl IntoResponse {
```
Comment (tasn) [inline server/svix-server/src/lib.rs:157]:
```suggestion
    async fn get_openapi_json() -> Result<Json> {
```

File: server/svix-server/src/lib.rs [hunk 3]
```diff
@@ -135,3 +136,27 @@ pub async fn run(cfg: Configuration, listener: Option<TcpListener>) {
...
+        let json: serde_json::Value =
+            serde_json::from_str(include_str!("static/openapi.json")).unwrap();
```
Comment (tasn) [inline server/svix-server/src/lib.rs:159]: Error handling.

File: server/svix-server/src/static/docs.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+<html>
+<head>
+<title>Svix API - ReDoc</title>
+<!-- needed for adaptive design -->
+<meta charset="utf-8"/>
+<meta name="viewport" content="width=device-width, initial-scale=1">
+
+<link href="https://fonts.googleapis.com/css?family=Montserrat:300,400,700|Roboto:300,400,700" rel="stylesheet">
+
+<link rel="shortcut icon" href="https://fastapi.tiangolo.com/img/favicon.png">
```
Comment (tasn) [inline server/svix-server/src/static/docs.html:11]: This is the fastapi icon... :)
You can just remove this line for now, or just point to something on svix.com.

File: server/svix-server/src/lib.rs [hunk 5]
```diff
@@ -135,3 +136,27 @@ pub async fn run(cfg: Configuration, listener: Option<TcpListener>) {
...
+
+mod docs {
+    use axum::{
+        response::{Html, IntoResponse},
+        routing::get,
+        Json, Router,
+    };
+
+    pub fn router() -> Router {
+        Router::new()
```
Comment (tasn) [inline server/svix-server/src/lib.rs:148]: Missing a redirect (302) from `/` to `/docs`.

**Review by tasn — COMMENTED** [review]
Looks great, just a few tiny comments.

File: server/svix-server/src/lib.rs [hunk 6]
```diff
@@ -135,3 +136,28 @@ pub async fn run(cfg: Configuration, listener: Option<TcpListener>) {
...
+mod docs {
+    use axum::{
+        response::{Html, IntoResponse, Redirect},
+        routing::get,
+        Json, Router,
+    };
+
+    pub fn router() -> Router {
+        Router::new()
+            .route("/", get(|| async { Redirect::permanent("/docs") }))
```
Comment (tasn) [inline server/svix-server/src/lib.rs:149]: 302, so not permanent, but temporary (who knows, we may want to do a different redirect).

File: server/svix-server/src/lib.rs (hunk 2 above)
Comment (svix-dylan) [reply server/svix-server/src/lib.rs:158]: Axum's Json requires a type signature :|

File: server/svix-server/src/static/docs.html [hunk 7]
```diff
@@ -8,7 +8,7 @@
 
 <link href="https://fonts.googleapis.com/css?family=Montserrat:300,400,700|Roboto:300,400,700" rel="stylesheet">
 
-<link rel="shortcut icon" href="https://fastapi.tiangolo.com/img/favicon.png">
+<link rel="shortcut icon" href="https://svix.com/favicon.ico">
```
Comment (tasn) [inline server/svix-server/src/static/docs.html:11]:
```suggestion
<link rel="shortcut icon" href="https://www.svix.com/favicon.ico">
```

File: server/svix-server/src/lib.rs (hunk 2 above)
Comment (tasn) [reply server/svix-server/src/lib.rs:158]: OK, it doesn't matter that much, but you can just use `Result<Json<Json::Value>>` or whatever it is.

**Review by tasn — APPROVED** [review]

**Pull request merged**